### PR TITLE
using Protobuf to enable Http experience receiving jobs' responses from Azure storage queue

### PR DESCRIPTION
**Pull request by homezcx**

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloDriver.cs [hunk 1]
```diff
@@ -48,7 +49,22 @@ private HelloDriver(IEvaluatorRequestor evaluatorRequestor)
         public void OnNext(IAllocatedEvaluator allocatedEvaluator)
         {
             _Logger.Log(Level.Info, "Evaluator allocated: {0}", allocatedEvaluator);
-            azureStorageQueueSender.SendMessageAsync("Reef/v1/driver/");
+
+            // TODO: Task 244286
+            AzureStorageHttpProxyConnection httpProxyConnection = new AzureStorageHttpProxyConnection(
+                "###", "###", "chzhareefrequestqueue");
+
+            HttpProxyResponseProto response = httpProxyConnection.RequestMessageResponseAsync(new HttpProxyRequestProto()
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloDriver.cs]: User will use such code to send the request and get response.

**Comment by homezcx** [comment]
This is mainly the .NET side change. Java side change will be handled in task 244273.

**Comment by homezcx** [comment]
Added Java side code.

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs [hunk 2]
```diff
@@ -1,52 +0,0 @@
...
-using System.Threading.Tasks;
-using Microsoft.WindowsAzure.Storage;
-using Microsoft.WindowsAzure.Storage.Queue;
-using Org.Apache.REEF.Utilities.Logging;
-
-namespace Org.Apache.REEF.Client.DotNet.AzureBatch
-{
-    public class AzureStorageQueueReceiver
-    {
-        private static readonly Logger LOGGER = Logger.GetLogger(typeof(AzureBatchDotNetClient));
```
Comment (sharathmalladi) [inline lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs]: typeof(AzureStorageQueueReceiver) instead.

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs [hunk 3]
```diff
@@ -1,52 +0,0 @@
...
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse("");
-
-            // Create the queue client.
-            CloudQueueClient queueClient = storageAccount.CreateCloudQueueClient();
-
-            // Retrieve a reference to a container.
-            queue = queueClient.GetQueueReference("responsequeue");
-        }
-
-        public void StartProcessingAsync()
```
Comment (sharathmalladi) [inline lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs]: Make the method async or remove the 'Async' in the method name.

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs [hunk 4]
```diff
@@ -1,52 +0,0 @@
...
-                    CloudQueueMessage message = await queue.GetMessageAsync();
-                    if(message != null)
-                    {
-                        await queue.DeleteMessageAsync(message);
-                    }
-                    else
-                    {
-                        continue;
-                    }
-                    using (StreamWriter w = File.AppendText("C:\\Users\\chzha\\Desktop\\log.txt"))
```
Comment (sharathmalladi) [inline lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs]: This is only writing the messages to a file and not processing them.

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs (hunk 3 above)
Comment (homezcx) [reply lang/cs/Org.Apache.REEF.Examples.HelloREEF/AzureStorageQueueReceiver.cs]: This code is in first commit and it has been removed.

File: lang/cs/Org.Apache.REEF.Client/AzureBatch/Parameters/AzureStorageHttpProxyDefaultResponseQueueName.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Client.AzureBatch.Parameters
+{
+    [NamedParameter(Documentation = "The Azure Batch Storage HTTP Proxy Default Response Queue", DefaultValue = "")]
+    public sealed class AzureStorageHttpProxyDefaultResponseQueueName : Name<string>
```
[the remark this answers is not in the document]
Comment (homezcx) [reply lang/cs/Org.Apache.REEF.Client/AzureBatch/Parameters/AzureStorageHttpProxyDefaultResponseQueueName.cs]: I will fix this. Using .DotNet.csproj is not required to add these files. I will add then for back compatibility in the other csproj file.

**Pull request closed without merging**